api/v1beta1/azuremachine_webhook_test.go

```diff
@@ -698,6 +698,21 @@ func TestAzureMachine_ValidateUpdate(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "validtest: azuremachine.spec.networkInterfaces[0].subnetName is overwritten when empty",
+			oldMachine: &AzureMachine{
+				Spec: AzureMachineSpec{
+					SubnetName:        "subnet1",
+					NetworkInterfaces: []NetworkInterface{{SubnetName: ""}},
+				},
+			},
+			newMachine: &AzureMachine{
+				Spec: AzureMachineSpec{
+					NetworkInterfaces: []NetworkInterface{{SubnetName: "subnet1"}},
+				},
+			},
+			wantErr: false,
```

Comment:
Apologies for not being clearer in the issue description, but it could help to add a check to ensure that the oldMachine's subnet name is set to the new machine's subnet name. To do this, you can add a boolean field like `checkSubnetName` to the test struct [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/6372a0514a5ea2345b3797695083c34c758bf8c9/api/v1beta1/azuremachine_webhook_test.go#L169). Then, you can check if the subnet name is equal when you [run the test](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/6372a0514a5ea2345b3797695083c34c758bf8c9/api/v1beta1/azuremachine_webhook_test.go#L718) if the boolean is set to true.

Reply:
Actually I think I may have misled you... since the subnet name is immutable, they should always be equal. Thus, the boolean is not needed and you can check for the equality of the subnet name for each test case.